component/src/main/java/org/wso2/extension/siddhi/store/influxdb/InfluxDBEventTable.java

```diff
@@ -0,0 +1,733 @@
+     *                                        such that the  system will take care retrying for connection
+     */
+    @Override
+    protected void connect() throws ConnectionUnavailableException {
+
+        try {
+            influxdb = InfluxDBFactory.connect(url, username, password);
+            if (!this.checkDatabaseExists(database)) {
+                connected = false;
+                throw new InfluxDBTableException("database " + database + " does not exist");
+            }
+            if (!InfluxDBTableUtils.isEmpty(retention)) {
+                influxdb.setRetentionPolicy(retention);
+            }
+            influxdb.setDatabase(database);
+            connected = true;
+        } catch (InfluxDBIOException e) {
+            connected = false;
+            throw new ConnectionUnavailableException("failed to initialize influxDB store " + e.getMessage(), e);
+        } catch (InfluxDBException.AuthorizationFailedException e) {
+            connected = false;
+            throw new InfluxDBTableException("wrong url or username " + e.getMessage(), e);
+        }
+    }
+
+    @Override
+    protected void disconnect() {
+
+        if (connected) {
+            this.influxdb.close();
+        }
+    }
+
+    @Override
+    protected void destroy() {
+
+        this.disconnect();
+    }
+
+    public QueryResult getSelectQueryResult(Map<String, Object> selectConditionParameterMap,
+                                            String condition, Map<Integer, Object> constantMap) {
+
+        String findCondition;
+        influxdb.setDatabase(database);
+        Query query;
+        QueryResult queryResult;
+        StringBuilder line = new StringBuilder();
+        line.append(SELECT_QUERY).append(this.tableName);
+        if (condition.equals("'*'")) {
+            findCondition = line.toString();
+        } else {
+            line.append(INFLUXQL_WHERE);
+            for (Map.Entry<String, Object> entry : selectConditionParameterMap.entrySet()) {
+                Object myObject = entry.getValue();
+                if (myObject instanceof String) {
+                    condition = condition.replaceFirst(Pattern.quote("?"), myObject.toString());
+                } else {
+                    condition = condition.replaceFirst(Pattern.quote("'?'"), myObject.toString());
+                }
+            }
+            for (Map.Entry<Integer, Object> entry : constantMap.entrySet()) {
+                Object myObject = entry.getValue();
+                if (myObject instanceof String) {
+                    condition = condition.replaceFirst(Pattern.quote("*"), myObject.toString());
+                } else {
+                    condition = condition.replaceFirst(Pattern.quote("'*'"), myObject.toString());
+                }
+            }
+            findCondition = line.toString();
+            findCondition = findCondition + condition;
```

Comment:
Use a string builder for findCondition. Also we don't need `line` and `condition` variables here. You can append to findCondition.